server/sonar-web/src/main/js/api/favorites.js

```diff
@@ -31,6 +31,7 @@ export function addFavorite (componentKey: string) {
 }
 
 export function removeFavorite (componentKey: string) {
-  const url = '/api/favourites/' + encodeURIComponent(componentKey);
-  return requestDelete(url);
+  const url = '/api/favorites/remove';
+  const data = { component: componentKey };
+  return post(url, data);
```

Comment:
I'd rename parameter to `component`, that'd allow to use more compact syntax `const data = { component }`. And finally I'd inline both variables to the call:
```
return post('/api/favorites/remove', { component });
```